cdap-ui/app/services/app-level-loading-service.js

```diff
@@ -38,4 +38,14 @@ angular.module(PKG.name + '.services')
       }
     };
 
+    this.hideLoadingIconImmediate = function(){
+      if (!deferred) {
+        return $q.when(true);
+      } else {
```

Comment:
After a return statement in the `if` block, no need to write `else`. Less indented code is easier to read.